### PR TITLE
Validate document id formating

### PR DESCRIPTION
**Pull request by bidoubiwa**
fixes: #104

In this PR we are checking if the document id of an entry is well formatted based on [Meilisearch requirements](https://docs.meilisearch.com/learn/core_concepts/primary_key.html#formatting-the-document-id). 

It also includes some changes on the logs, in order to log the document_id error we added, but also to avoid showing an entry contained in the logs. For security reasons but also to avoid logs being too big.

### CODE REVIEW

File: functions/src/index.ts [hunk 1]
```diff
@@ -77,8 +85,14 @@ async function handleAddDocument(
  */
 async function handleDeleteDocument(documentId: string): Promise<void> {
   try {
-    await index.deleteDocument(documentId)
     logs.deleteDocument(documentId)
+    if (validateDocumentId(documentId)) {
+      await index.deleteDocument(documentId)
+    } else {
+      logger.error(
+        `Could not create document with id: ${documentId}. The document id can only contain case-insensitive alphanumeric characters (abcDEF), hyphens (-) or underscores(_).`
```
Comment (alallema) [inline functions/src/index.ts:93]: Shouldn't it be something like: `Could not delete document` or `Document can't exist with id: ...`

File: functions/src/index.ts [hunk 2]
```diff
@@ -94,9 +108,15 @@ async function handleUpdateDocument(
...
-    const document = adaptDocument(documentId, after)
-    await index.updateDocuments([document])
-    logs.updateDocument(documentId, document)
+    logs.updateDocument(documentId)
+    if (validateDocumentId(documentId)) {
+      const document = adaptDocument(documentId, after)
+      await index.updateDocuments([document])
+    } else {
+      logger.error(
+        `Could not create document with id: ${documentId}.The document id can only contain case-insensitive alphanumeric characters (abcDEF), hyphens (-) or underscores(_).`
```
Comment (alallema) [inline functions/src/index.ts:117]: Sam here for `Could not create`

**Review by alallema — DISMISSED** [review]
Great! 🌯

**Review by alallema — APPROVED** [review]
LGTM! 🎉

**Comment by bidoubiwa** [comment]
bors merge

**Pull request merged**